src/rlookup.c

```diff
@@ -106,10 +106,12 @@ RLookupKey *RLookup_GetKey(RLookup *lookup, const char *name, int flags) {
   return RLookup_GetKeyEx(lookup, name, strlen(name), flags);
 }
 
-size_t RLookup_GetLength(const RLookup *lookup, const RLookupRow *r, int requiredFlags,
-                         int excludeFlags) {
+size_t RLookup_GetLength(const RLookup *lookup, const RLookupRow *r, int *skipFieldIndex,
+                         int requiredFlags, int excludeFlags, SchemaRule *rule) {
+  int i = -1; // we do i++ right away and we don't want to skip i == 0
   size_t nfields = 0;
   for (const RLookupKey *kk = lookup->head; kk; kk = kk->next) {
```

Comment:
You can do:
`for (const RLookupKey *kk = lookup->head, i = 0; kk; kk = kk->next, ++i)`
IIRC

Reply:
Nice touch!